### PR TITLE
refactor(tags): migration typescript

### PR DESCRIPTION
**Pull request by philibea**
## Type

- Refactor Tags Component to Typescript

### CODE REVIEW

File: src/components/Tags/__tests__/index.tsx [hunk 1]
```diff
@@ -114,8 +116,10 @@ describe('Tags', () => {
       {
         transform: async () => {
           const input = document.getElementById('test')
```
Comment (DorianMaliszewski) [inline src/components/Tags/__tests__/index.tsx:118]: We need to drop all document.X and use jest selectors. Our tests mustn't be conditonnals @philibea

File: src/components/Tags/index.tsx [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      setStatus({ [tagIndex]: STATUS.IDLE })
+    } catch (e) {
+      onChangeError?.(e)
+      setTags(tagsState)
+    }
+  }
+
+  const handleInputKeydown = e => {
+    // 32 = Space | 13 = Enter | 8 = backspace
+    if (e.keyCode === 32 || e.keyCode === 13) {
```
Comment (DorianMaliszewski) [inline src/components/Tags/index.tsx:189]: keyCode is deprecated, is it possible to change it in this PR ?

File: src/components/Tags/index.tsx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+      addTag()
+      e.preventDefault()
+    }
+    if (
+      e.keyCode === 8 &&
```
Comment (DorianMaliszewski) [inline src/components/Tags/index.tsx:194]: Same

**Review by DorianMaliszewski — CHANGES_REQUESTED** [review]

File: src/components/Tags/index.tsx (hunk 2 above)
Comment (philibea) [reply src/components/Tags/index.tsx]: i made something for this component, but i believe we should probably use https://github.com/timoxley/keycode

File: src/components/Tags/__tests__/index.tsx (hunk 1 above)
Comment (philibea) [reply src/components/Tags/__tests__/index.tsx]: Solved

**Review by DorianMaliszewski — APPROVED** [review]

File: src/components/Tags/index.tsx [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+      : tagsState
+    setInput('')
+    setTags(newTags)
+    if (newTags.length !== tagsState.length) {
+      setStatus({ [newTags[newTags.length - 1].index]: STATUS.LOADING })
+    }
+    try {
+      dispatchOnChange(newTags)
+      setStatus({ [newTags[newTags.length - 1].index]: STATUS.IDLE })
+    } catch (error) {
```
Comment (chambo-e) [inline src/components/Tags/index.tsx:174]: ERROR

**Review by chambo-e — APPROVED** [review]

**Pull request merged**